src/sdp_proxy/advertising_proxy.cpp

```diff
@@ -180,6 +180,31 @@ void AdvertisingProxy::OnMdnsPublishResult(otSrpServerServiceUpdateId aUpdateId,
     }
 }
 
+std::vector<Ip6Address> AdvertisingProxy::GetEligibleAddresses(const otIp6Address *aHostAddresses,
+                                                               uint8_t             aHostAddressNum)
+{
+    std::vector<Ip6Address> addresses;
+    const otIp6Address *    meshLocalEid = otThreadGetMeshLocalEid(GetInstance());
+
+    addresses.reserve(aHostAddressNum);
+    for (size_t i = 0; i < aHostAddressNum; ++i)
+    {
+        Ip6Address address(aHostAddresses[i].mFields.m8);
+
+        if (otIp6PrefixMatch(meshLocalEid, &aHostAddresses[i]) >= OT_IP6_PREFIX_BITSIZE)
+        {
+            continue;
+        }
+        if (address.IsLinkLocal())
+        {
+            continue;
+        }
```

Comment:
LLA was removed in SRP server. Not sure if it is still the correct behavior per the latest Thread spec.

Reply:
It should be removed, right?
I guess it's fine to remove it twice.